cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -427,6 +428,24 @@ std::shared_ptr<Table> TableFromJSON(const std::shared_ptr<Schema>& schema,
   return *Table::FromRecordBatches(schema, std::move(batches));
 }
 
+Result<std::shared_ptr<Table>> RunEndEncodeTableColumns(
+    const Table& table, const std::vector<int>& column_indices) {
+  const int num_columns = table.num_columns();
+  std::vector<std::shared_ptr<ChunkedArray>> encoded_columns;
+  encoded_columns.reserve(num_columns);
+  for (int i = 0; i < num_columns; i++) {
+    if (std::find(column_indices.begin(), column_indices.end(), i) !=
+        column_indices.end()) {
+      EXPECT_OK_AND_ASSIGN(auto run_end_encoded, compute::RunEndEncode(table.column(i)));
+      EXPECT_EQ(run_end_encoded.kind(), Datum::CHUNKED_ARRAY);
```

Comment:
Given you're returning a result anyways you could just use ARROW_ASSIGN_OR_RAISE and ARROW_RETURN_NOT_OK here right?

Reply:
Isn't this better for things used in tests? These are not really expected to fail, so these checks are more assert-like than regular error checking. I found this to be a pattern in this file.

Reply:
If that's the case then change the return type to `std::shared_ptr<Table>` and change the end to...

```
EXPECT_OK_AND_ASSIGN(auto table, Table::Make(...));
return table;
```

I don't really mind one pattern or the other.  It's just the mixing of patterns we should avoid.

Reply:
Done. Pushed two new commits.